### DISCUSSION TITLE
Just an idea

### DISCUSSION THREAD

**Discussion by randomwangran** [karthink/gptel#118] [Ideas]
https://github.com/karthink/gptel/blob/e61180ebc59758a2298ae2441b2fd102d80ec2eb/gptel.el#L376

If  TODO -> DONE,  things like sectional suggestions can be achieved:

https://youtu.be/MpK9PWo0lUw?t=115

**Comment by randomwangran** [comment]
#15 related

**Comment by karthink** [comment]
> https://youtu.be/MpK9PWo0lUw?t=115

I'm not sure how this feature connects with the TODO item the code. Could you explain in a little more detail?

**Comment by psionic-k** [comment]
Fundamentally, since we can only edit on one dimension of a request at a time, I think what they mean is to map the user's command over either a list of data or using a list of contexts, prompts, or other request dimensions.

Multiple requests only makes sense when the things being mapped over do not provide context to each other or the goal is to eliminate context leakage from item to item.  If the structure of output is difficult to generate correctly, that's another possibility.

If anything is needed, it's probably just the infrastructure to support multiple simultaneous requests.  A plural API would be convenient, but really just front-loading some of the plumbing that users can / should do on their own.

**Comment by karthink** [comment]
> If anything is needed, it's probably just the infrastructure to support multiple simultaneous requests.

gptel does not currently enforce any limit on the number of simultaneous requests (in a single buffer or across buffers), so you can fire off as many as you want.  I often have 2 requests going at once.

- What is a "plural API"?
- By "infrastructure" do you mean a UI?

**Comment by psionic-k** [comment]
infrastructure: the Open AI API calling implementation code.  (I didn't look at it yet)
plural API: one that handles lists of things instead of a thing
